src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunSingleAssemblyCompilationModuleGroup.cs

```diff
@@ -14,26 +14,25 @@ namespace ILCompiler
     public class ReadyToRunSingleAssemblyCompilationModuleGroup : CompilationModuleGroup
     {
         private HashSet<ModuleDesc> _compilationModuleSet;
+        private HashSet<ModuleDesc> _versionBubbleModuleSet;
 
-        public ReadyToRunSingleAssemblyCompilationModuleGroup(TypeSystemContext context, IEnumerable<ModuleDesc> compilationModuleSet)
+        public ReadyToRunSingleAssemblyCompilationModuleGroup(
+            TypeSystemContext context, 
+            IEnumerable<ModuleDesc> compilationModuleSet,
+            IEnumerable<ModuleDesc> versionBubbleModuleSet)
         {
             _compilationModuleSet = new HashSet<ModuleDesc>(compilationModuleSet);
 
             // The fake assembly that holds compiler generated types is part of the compilation.
             _compilationModuleSet.Add(context.GeneratedAssembly);
+
+            _versionBubbleModuleSet = new HashSet<ModuleDesc>(versionBubbleModuleSet);
+            _versionBubbleModuleSet.UnionWith(_compilationModuleSet);
         }
 
         public sealed override bool ContainsType(TypeDesc type)
         {
-            if (type is EcmaType ecmaType)
-            {
-                return IsModuleInCompilationGroup(ecmaType.EcmaModule);
-            }
-            if (type is InstantiatedType instantiatedType)
-            {
-                return ContainsType(instantiatedType.GetTypeDefinition());
-            }
-            return true;
+            return type.GetTypeDefinition() is EcmaType ecmaType && IsModuleInCompilationGroup(ecmaType.EcmaModule);
```

Comment:
Where did the check of the `Instantiation` go?

Reply:
I belive this is just a simplification of the previous formulation - we call GetTypeDefinition upfront instead of recursively calling ContainsType.

Reply:
ok